homeassistant/components/powerwall/__init__.py

```diff
@@ -183,44 +179,81 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return True
 
 
-def _login_and_fetch_base_info(
+async def _login_and_fetch_base_info(
     power_wall: Powerwall, host: str, password: str | None
 ) -> PowerwallBaseInfo:
     """Login to the powerwall and fetch the base info."""
     if password is not None:
-        power_wall.login(password)
-    return call_base_info(power_wall, host)
+        await power_wall.login(password)
+    return await _call_base_info(power_wall, host)
 
 
-def call_base_info(power_wall: Powerwall, host: str) -> PowerwallBaseInfo:
+async def _call_base_info(power_wall: Powerwall, host: str) -> PowerwallBaseInfo:
     """Return PowerwallBaseInfo for the device."""
-    # Make sure the serial numbers always have the same order
-    gateway_din = None
-    with contextlib.suppress(AssertionError, PowerwallError):
-        gateway_din = power_wall.get_gateway_din().upper()
+
+    async def get_gateway_din() -> str:
+        # Make sure the serial numbers always have the same order
+        gateway_din: str
+        with contextlib.suppress(AssertionError, PowerwallError):
+            gateway_din = await power_wall.get_gateway_din()
+
+        return gateway_din.upper()
+
+    (
+        gateway_din,
+        site_info,
+        status,
+        device_type,
+        serial_numbers,
+    ) = await asyncio.gather(
+        get_gateway_din(),
+        power_wall.get_site_info(),
+        power_wall.get_status(),
+        power_wall.get_device_type(),
+        power_wall.get_serial_numbers(),
+    )
+
     return PowerwallBaseInfo(
         gateway_din=gateway_din,
-        site_info=power_wall.get_site_info(),
-        status=power_wall.get_status(),
-        device_type=power_wall.get_device_type(),
-        serial_numbers=sorted(power_wall.get_serial_numbers()),
+        site_info=site_info,
+        status=status,
+        device_type=device_type,
+        serial_numbers=serial_numbers,
```

Comment:
```suggestion
        serial_numbers=sorted(serial_numbers),
```

These must be sorted or the unique id can change each time

Reply:
Seems like this should be something that the user of this library does since it isn't really this library's requirement?